File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // yell if user doesn't give single command line argument
    if (argc != 2)
    {
        printf("You should provide a command line argument!");
        return 1;
    }
    
    // convert string from command line argument to integer key
    int key = atoi(argv[1]);
    
    // get user input
    string text = GetString();
    
    // encrypt message with caesar cipher
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // encryption for uppercase letter
        if (isalpha(text[i]) && isupper(text[i]))
        {
            printf("%c", ((text[i] - 'A' + key) % 26) + 'A');
        }
        // encryption for lowercase letter
        else if (isalpha(text[i]) && islower(text[i]))
        {
            printf("%c", ((text[i] - 'a' + key) % 26) + 'a');
        }
        // otherwise, just ignore it without encryption
        else
        {
            printf("%c", text[i]);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("You should provide cmd line arguments!")
        exit(1)
    
    key = int(sys.argv[1])
    translated = []
    message = cs50.get_string()
    for symbol in message:
        if symbol.isalpha():
            translated.append(caesar(symbol, key))
        else:
            translated.append(symbol)
                
    print("".join(translated))
    exit(0)

def caesar(char, key):
    if char.isupper():
        return chr(((ord(char) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(char) - 97 + key) % 26) + 97)
        
if __name__ == "__main__":
    main()